components/brave_ads/core/internal/account/utility/refill_confirmation_tokens/refill_confirmation_tokens.cc

```diff
@@ -272,29 +274,29 @@ void RefillConfirmationTokens::ParseAndRequireCaptcha(
 }
 
 void RefillConfirmationTokens::SuccessfullyRefilled() {
-  is_refilling_ = false;
-
   StopRetrying();
 
   Reset();
 
+  is_refilling_ = false;
+
   NotifyDidRefillConfirmationTokens();
 }
 
-void RefillConfirmationTokens::FailedToRefill(const bool should_retry) {
-  is_refilling_ = false;
+void RefillConfirmationTokens::FailedToRefillAndRetry() {
+  NotifyFailedToRefillConfirmationTokens();
 
-  if (!should_retry) {
-    StopRetrying();
+  Retry();
+}
 
-    Reset();
-  }
+void RefillConfirmationTokens::FailedToRefill() {
+  StopRetrying();
 
-  NotifyFailedToRefillConfirmationTokens();
+  Reset();
 
-  if (should_retry) {
-    Retry();
-  }
+  is_refilling_ = false;
```

Comment:
Maybe put `is_refilling_ = false;` in `Reset()`? You seem to set it to `false` after every call to `Reset()`.

Reply:
Thanks, changed